lib/course_planner_web/models/bulk_helper.ex

```diff
@@ -0,0 +1,78 @@
+defmodule CoursePlanner.BulkHelper do
+  @moduledoc """
+    Handle bulk creation specific logics
+  """
+  alias CoursePlanner.{Repo, Accounts.User}
+  alias Coherence.ControllerHelpers
+  alias Ecto.{DateTime, Multi}
+
+  def bulk_user_creation(csv_file_stream) do
+    csv_parsed_result = csv_parsing(csv_file_stream)
+
+    case validate_column_count(csv_parsed_result, 5) do
+      {:ok, parsed_params} ->
+        create_users_with_transaction(parsed_params)
+      {:validate_error, row_index} ->
+        {:error, "parsing_csv",
+          "Input data in row ##{row_index + 1} is not matching the column number.", ""}
+      {:error, :empty} ->
+        {:error, "parsing_csv", "Input can not be empty.", ""}
+      {:error, message} ->
+        {:error, "parsing_csv", message, ""}
+    end
+  end
+
+  defp csv_parsing(csv_file_stream) do
+    csv_file_stream
+    |> CSV.decode
```

Comment:
I think it would be more interesting to call `CSV.decode(strip_fields: true, headers: %w(name, family_name, nickname, email, role)a)`

this will tell the library the number of columns you are expecting already, besides that it creates already maps for each line, and it will strip the columns